runtime/src/topo/hwloc/topo-hwloc.c

```diff
@@ -517,6 +517,27 @@ void alignAddrSize(void* p, size_t size, chpl_bool onlyInside,
 }
 
 
+void chpl_topo_interleaveMemLocality(void* p, size_t size) {
+  int flags;
+
+  if (!haveTopology) {
+    return;
+  }
+
+  if (!topoSupport->membind->set_area_membind) {
+    return;
+  }
+
+  hwloc_bitmap_t set;
+  hwloc_obj_t obj;
+  obj = hwloc_get_root_obj(topology);
+  set = hwloc_bitmap_dup(obj->cpuset);
+
+  flags = HWLOC_MEMBIND_MIGRATE | HWLOC_MEMBIND_STRICT;
```

Comment:
For sure do you want both of these flags? `MIGRATE` says to physically move the pages if they already exist and don't match the desired locality. Doing so would take a while if there were a lot of memory involved, and would break things if the memory were already registered with either a NC or a GPU. It might not matter right now, because there's only one caller of this routine and we expect the pages don't previously exist, right? But if there are other callers someday then I'd think we might not want to move the memory in all cases. We could deal with that when the need arose, I guess.
`STRICT` says to return error if the policy can't be set (nonexistent pages) or the memory cannot be moved (existent pages). Maybe it would be better to warn in that case instead of killing off the app altogether?

Reply:
I think you're right that we don't want them, I just copied these from `chpl_topo_setMemLocalityByPages`. 

I tested this code path with gasnet-ibv-fast and memory will already be faulted in there, so I'm surprised that didn't cause any issues.

Will open up a followup PR to remove these flags

Reply:
> I tested this code path with gasnet-ibv-fast and memory will already be faulted in there, so I'm surprised that didn't cause any issues.

As long as it actually could migrate any pages it needed to this wouldn't have caused an error. But if the amount of memory that had to be moved were large (multi-G, say) the time should have been noticeable.

Reply:
There was no noticeable time. We already interleave the heap manually at startup for gasnet-ibv-fast, so I wonder if our interleaving happened to match what hwloc would do anyways.

Reply:
Done in https://github.com/chapel-lang/chapel/pull/18383